src/transformers/feature_extraction_sequence_utils.py

```diff
@@ -342,9 +345,20 @@ def _truncate(
         needs_to_be_truncated = len(required_input) > max_length
 
         if needs_to_be_truncated:
-            processed_features[self.model_input_names[0]] = processed_features[self.model_input_names[0]][:max_length]
-            if "attention_mask" in processed_features:
-                processed_features["attention_mask"] = processed_features["attention_mask"][:max_length]
+            if self.truncation_side == "right":
+                processed_features[self.model_input_names[0]] = processed_features[self.model_input_names[0]][
+                    :max_length
+                ]
+                if "attention_mask" in processed_features:
+                    processed_features["attention_mask"] = processed_features["attention_mask"][:max_length]
+            elif self.truncation_side == "left":
+                processed_features[self.model_input_names[0]] = processed_features[self.model_input_names[0]][
+                    max_length:
```

Comment:
What happens here actually if `max_length` is larger then one input? E.g. imagine we have the following inputs:

`[[1, 2, 3, 4, 5], [1, 2, 3], [1, 2]]` and we want to truncate & pad to a length of 3 => with `truncation_side == "left"` this fails since here: https://github.com/huggingface/transformers/blob/3ff2cde5ca4a2d3c622b827d9edf7e3d0b7f4fb7/src/transformers/feature_extraction_sequence_utils.py#L214 we would truncate the last list `[1, 2]` essentially to an empty list no?

Reply:
Actually, I think we should just change it to `[-max_length:]` and it should work

Reply:
Fixed this.